metrics-facade-spring/metrics-facade-spring-boot-sample-app/src/main/java/com/ringcentral/platform/metrics/spring/CounterEndpoint.java

```diff
@@ -2,22 +2,51 @@
 
 import com.ringcentral.platform.metrics.MetricRegistry;
 import com.ringcentral.platform.metrics.counter.Counter;
-import org.springframework.web.bind.annotation.*;
+import com.ringcentral.platform.metrics.counter.Counter.Count;
+import com.ringcentral.platform.metrics.samples.prometheus.PrometheusInstanceSampleSpecModsProvider;
+import com.ringcentral.platform.metrics.samples.prometheus.PrometheusSampleSpecModsProvider;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
 
 import static com.ringcentral.platform.metrics.names.MetricName.withName;
+import static com.ringcentral.platform.metrics.names.MetricNameMask.forMetricWithName;
+import static com.ringcentral.platform.metrics.samples.prometheus.PrometheusSampleSpec.sampleSpec;
+import static java.lang.Double.parseDouble;
 import static org.springframework.web.bind.annotation.RequestMethod.POST;
 
 @RestController
 public class CounterEndpoint {
 
     private final Counter counter;
+    private final PrometheusInstanceSampleSpecModsProvider instanceSampleSpecModsProvider;
+    private final PrometheusSampleSpecModsProvider sampleSpecModsProvider;
+
+    public CounterEndpoint(
+        MetricRegistry metricRegistry,
+        PrometheusInstanceSampleSpecModsProvider instanceSampleSpecModsProvider,
+        PrometheusSampleSpecModsProvider sampleSpecModsProvider) {
 
-    public CounterEndpoint(MetricRegistry metricRegistry) {
         this.counter = metricRegistry.counter(withName("counter"));
+        this.instanceSampleSpecModsProvider = instanceSampleSpecModsProvider;
+        this.sampleSpecModsProvider = sampleSpecModsProvider;
     }
 
     @RequestMapping(value = "/counter/inc", method = POST)
     public synchronized void incCounter() {
         counter.inc();
     }
+
+    @RequestMapping(value = "/counter/prometheus/value/{value}", method = POST)
```

Comment:
I'd suggest adding a removal method to make the example more realistic.

Reply:
ok

Reply:
done